indico/util/spreadsheets.py

```diff
@@ -86,17 +88,20 @@ def generate_csv(headers, rows):
     return buf
 
 
-def _prepare_excel_data(data):
+def _prepare_excel_data(data, tz=None):
     if isinstance(data, (list, tuple)):
         data = ', '.join(data)
     elif isinstance(data, set):
         data = ', '.join(sorted(data, key=unicode.lower))
     elif is_lazy_string(data) or isinstance(data, Markup):
         data = unicode(data)
+    elif isinstance(data, datetime):
+        data = format_datetime(data, timezone=tz) if tz else format_datetime(data)
```

Comment:
No need for the if..else, `timezone=None` is the default so you can just go for `data = format_datetime(data, timezone=tz).decode('utf-8')` without any extra checks